### ISSUE TITLE
Error in X_chosen[data$obsID, ] : subscript out of bounds

### ISSUE THREAD

**Issue by Ales-G** [jhelvy/logitr#30]
Hello, 
it is me again!
apologies for all these comments but I really like your program and I am using it on a number of datasets, which makes me encounter a number of errors.

I am estimating a basic model, without clustered standard errors, I have a dataset of >3500 observations.

```
mnl_pref <- logitr(
+   data           = dat_transformed,
+   outcome        = "choice",
+   obsID          = "tskID",
+   pars           = c("wg",myvars),
+   modelSpace     = "pref",
+   panelID        = "wrkid",
+   numMultiStarts = 100)
# and I get the following error
Error in X_chosen[data$obsID, ] : subscript out of bounds
```
Again I have looked at my `tskID` variable, and it looks correct. 

Any idea of what may be causing this error?
thanks a lot for all of your help and support! you are really making a great contribution to me and to the community in general!

**Comment by jhelvy** [comment]
Yeah this too looks similar to the other error. Anytime you see an error with `obsID` involved, it usually has to do with how the IDs are set up. It's on my list to have a data check function be called right before estimating the model to make sure all the inputs are correct. It's hard to know what's causing this without seeing the data.

**Comment by jhelvy** [comment]
Btw, you may also want to try using [xlogit](https://github.com/arteagac/xlogit). It's a python package with a very similar UI to logitr (and it's actually even faster!). It doesn't have WTP space models yet, though that's in the works. It requires the same data structure, so if you use it and don't get any errors, that means there's a bug in logitr. If you get errors, then it's probably an error in the data somewhere.

**Comment by HeniCha** [comment]
Dear Professor Helveston, 
I am also a big fan of your great logitr code - especially because of the WTP space estimation! Thanks a lot for your work! 

I've been working with the code back in the beginning of May and I successfully ran it in preference and WTP space with and without clustered standard errors. 
Today, I came back working on my analyses and wanted to re-run the exact same code with the exact same dataset as I did in the beginning of May. I just updated the package to your new logitr 0.7.0 version. 
However, now I receive the same error Ales-G mentioned in this comment: 
Error in X_chosen[data$obsID, ] : subscript out of bounds

As I said, I haven't changed the dataset or the code since the beginning of May (except for changing the argument name price to scalePar and taking the modelSpace argument out). Even with the simplest model in preference space without clusters, I receive the error. I also checked my dataset again, but I always have four observations per obsID like it should be. 

Could it be that anything changed in the code that causes the error since the 0.7.0 update?  

I would very much appreciate your help and any ideas! Thanks a lot in advance!

**Comment by HeniCha** [comment]
Thanks so much for your quick reply! Please find attached the code:

````
sample <- read.csv("Merged_long_adj.csv")

attributes <- c("Dlabel2", "Dorigin", "asc")

set.seed(111)
pref_base <- logitr(
  data      = sample,
  outcome   = "response",
  obsID     = "gid",
  panelID   = "respid",
  clusterID = "respid", 
  numDraws  = 1000,
  pars      = c("price", attributes),
  randPars  = c(Dlabel2="n", Dorigin="n", asc="n")
)
````

The example is with clustered se, but even if I take panelID and clusterID out, I receive the error. 

Thanks so much!

**Comment by HeniCha** [comment]
Thanks a lot! It now works perfectly and I can successfully run the code again. 

Thanks so much for quickly fixing the errors and for your prompt reply!

**Comment by jhelvy** [comment]
That's great! @Ales-G any chance this also fixes the issue you were having?

**Comment by jhelvy** [comment]
I think this issue is now addressed with recent bug fixes in v0.7.2.
